pkg/sql/opt/xform/testdata/rules/select

```diff
@@ -6694,21 +6694,25 @@ project
  └── distinct-on
       ├── columns: k:1!null u:2!null v:3
       ├── grouping columns: k:1!null
+      ├── internal-ordering: +1 opt(2)
```

Comment:
This is now a streaming DistinctOn, which is better. However, notice that the DistinctOn should really be eliminated because it's grouping by `(1)` which is a strict key in the input. The DistinctOn is not eliminated because the normalization rule that should do it is not applied at the root of the explored expression. Notice [here](https://github.com/cockroachdb/cockroach/blob/d7a553b82237d5c2c8be7d109d2a3e9b1d69144a/pkg/sql/opt/xform/testdata/rules/select#L8000-L8023) how the `DistinctOn` is eliminated when SplitDisjunctionAddKey is applied, because of the addition Project it adds.